src/linear_algebra/contraction.lean

```diff
@@ -51,4 +60,45 @@ variables {R M N}
   dual_tensor_hom R M N (f ⊗ₜ n) m = (f m) • n :=
 by { dunfold dual_tensor_hom, rw uncurry_apply, refl, }
 
+/-- As a matrix, `dual_tensor_hom` evaluated on a basis element of `M* ⊗ N` is a matrix with a
+single one and zeros elsewhere -/
+theorem dual_tensor_hom_basis
+  {m : Type*} {n : Type*} [fintype m] [fintype n] [decidable_eq m] [decidable_eq n]
+  (bM : basis m R M) (bN : basis n R N) (j : m) (i : n) :
+    to_matrix bM bN (dual_tensor_hom R M N (bM.coord j ⊗ₜ bN i)) = std_basis_matrix i j 1 :=
+begin
+  ext i' j',
+  by_cases hij : (i = i' ∧ j = j');
+  simp [linear_map.to_matrix_apply, finsupp.single_eq_pi_single, hij],
+  rw [and_iff_not_or_not, not_not] at hij, cases hij; simp [hij],
+end
+
+end
+
+open finite_dimensional
+variables (R : Type u) (M N : Type v)
+variables [field R] [add_comm_group M] [add_comm_group N] [module R M] [module R N]
+variables [finite_dimensional R M]
+
+theorem dual_tensor_hom_surj : function.surjective (dual_tensor_hom R M N) :=
+begin
+  intro f,
+  have b := fin_basis R M,
+  use ∑ (i : fin (finrank R M)), (b.dual_basis i) ⊗ₜ f (b i),
```

Comment:
This looks like there might be an explicit right-inverse hiding here given a basis; can you extract it into a def?

Reply:
Done, and I proved that this explicit inverse is equal to `dual_tensor_hom_equiv.symm`.